src/common/exif.cc

```diff
@@ -741,9 +741,32 @@ static gboolean _check_dng_opcodes(Exiv2::ExifData &exifData, dt_image_t *img)
     g_free(data);
     has_opcodes = TRUE;
   }
-  else
+  return has_opcodes;
+}
+
+static gboolean _check_dng_opcodes3(Exiv2::ExifData &exifData, dt_image_t *img)
+{
+  gboolean has_opcodes = FALSE;
+  Exiv2::ExifData::const_iterator pos = exifData.findKey(Exiv2::ExifKey("Exif.SubImage1.OpcodeList3"));
+  // DNGs without an embedded preview have the opcodes under Exif.Image instead of Exif.SubImage1
+  if(pos == exifData.end())
+    pos = exifData.findKey(Exiv2::ExifKey("Exif.Image.OpcodeList3"));
+  if(pos != exifData.end())
   {
-    dt_vprint(DT_DEBUG_IMAGEIO, "DNG OpcodeList2 tag not found\n");
+    uint8_t *data = (uint8_t *)g_malloc(pos->size());
+    pos->copy(data, Exiv2::invalidByteOrder);
+    dt_dng_opcode_process_opcode_list_3(data, pos->size(), img);
+    g_free(data);
+    has_opcodes = TRUE;
+
+    // we also want the active area for a lens scaling calculation
+    img->exif_correction_data.dng.activearea[0] = -1;
+    Exiv2::ExifData::const_iterator posa = exifData.findKey(Exiv2::ExifKey("Exif.SubImage1.0xc68d"));
```

Comment:
FYI, 0xc68d has been around in exiv2 as "ActiveArea" for a while now (at least since 0.25), should be able to count on it...

Also, there is no guarantee the IFD location of this tag is in `Exif.SubImage1` - there are plenty of DNGs with no thumbnail so the raw IFD is actually the root `Exif.Image`, but in theory (although rarely seen), could be any sub-IFD location...